packages/core-flows/src/definition/cart/workflows/refresh-payment-collection.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  StepResponse,
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common/steps/use-remote-query"
+import {
+  deletePaymentSessionStep,
+  updatePaymentCollectionStep,
+} from "../../payment-collection"
+
+type WorklowInput = {
+  cart_id: string
+}
+
+interface StepInput {
+  cart_id: string
+}
+
+// We export a step running the workflow too, so that we can use it as a subworkflow e.g. in the update cart workflows
+export const refreshPaymentCollectionForCartStepId =
+  "refresh-payment-collection-for-cart"
+export const refreshPaymentCollectionForCartStep = createStep(
+  refreshPaymentCollectionForCartStepId,
+  async (data: StepInput, { container }) => {
+    await refreshPaymentCollectionForCartWorkflow(container).run({
+      input: {
+        cart_id: data.cart_id,
+      },
+    })
+
+    return new StepResponse(null)
+  }
+)
+
+export const refreshPaymentCollectionForCartWorkflowId =
+  "refresh-payment-collection-for-cart"
+export const refreshPaymentCollectionForCartWorkflow = createWorkflow(
+  refreshPaymentCollectionForCartWorkflowId,
+  (input: WorkflowData<WorklowInput>): WorkflowData<void> => {
+    const test = transform({ input }, (data) => {
+      console.log("Workflow >>>: ", data.input)
+    })
+
+    const carts = useRemoteQueryStep({
+      entry_point: "cart",
+      fields: [
+        "id",
+        "total",
+        "currency_code",
+        "payment_collection.id",
+        "payment_collection.payment_sessions.id",
+      ],
+      variables: { id: input.cart_id, test },
+    })
+
+    deletePaymentSessionStep({
+      payment_session_id: carts[0].payment_collection.payment_sessions?.[0].id,
```

Comment:
🙏 it is super cool to get this works knowing that at that time the carts does not even exists in itself but it is resolved in the future :)

Reply:
Yes, indeed - super cool!